test/StringUtils.test.cpp

```diff
@@ -116,60 +129,83 @@ TEST(String, stringFrom)
-
-	EXPECT_EQ("-1"s, NAS2D::stringFrom<int>(-1));
-	EXPECT_EQ("0"s, NAS2D::stringFrom<int>(0));
-	EXPECT_EQ("1"s, NAS2D::stringFrom<int>(1));
-
-	EXPECT_EQ("-1"s, NAS2D::stringFrom<long>(-1));
-	EXPECT_EQ("0"s, NAS2D::stringFrom<long>(0));
-	EXPECT_EQ("1"s, NAS2D::stringFrom<long>(1));
-
-	EXPECT_EQ("-1"s, NAS2D::stringFrom<long long>(-1));
-	EXPECT_EQ("0"s, NAS2D::stringFrom<long long>(0));
-	EXPECT_EQ("1"s, NAS2D::stringFrom<long long>(1));
-
-	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<float>(-1));
-	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<float>(-1.0));
-	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<float>(-1.1f));
-	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<float>(0));
-	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<float>(0.0));
-	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<float>(1));
-	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<float>(1.0));
-	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<float>(1.1f));
-
-	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<double>(-1));
-	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<double>(-1.0));
-	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<double>(-1.1));
-	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<double>(0));
-	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<double>(0.0));
-	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<double>(1));
-	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<double>(1.0));
-	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<double>(1.1));
-
-	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<long double>(-1));
-	EXPECT_EQ("-1.000000"s, NAS2D::stringFrom<long double>(-1.0));
-	EXPECT_EQ("-1.100000"s, NAS2D::stringFrom<long double>(-1.1L));
-	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<long double>(0));
-	EXPECT_EQ("0.000000"s, NAS2D::stringFrom<long double>(0.0));
-	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<long double>(1));
-	EXPECT_EQ("1.000000"s, NAS2D::stringFrom<long double>(1.0));
-	EXPECT_EQ("1.100000"s, NAS2D::stringFrom<long double>(1.1L));
+}
+
+
+template<typename T>
+class StringFromSignedIntegerTest : public ::testing::Test
+{
+  public:
+	  StringFromSignedIntegerTest<T>()
+	{}
+
+	T negOne{-1};
+	T zero{0};
+	T one{1};
+};
+
+using StringFromSignedIntegerTestTypes = ::testing::Types<short, int, long, long long>;
+TYPED_TEST_SUITE(StringFromSignedIntegerTest, StringFromSignedIntegerTestTypes);
+
+TYPED_TEST(StringFromSignedIntegerTest, StringFromSignedIntegerTestEdgeCases)
+{
+	using namespace std::literals::string_literals;
+
+	EXPECT_EQ("-1"s, NAS2D::stringFrom<TypeParam>(this->negOne));
+	EXPECT_EQ("0"s, NAS2D::stringFrom<TypeParam>(this->zero));
+	EXPECT_EQ("1"s, NAS2D::stringFrom<TypeParam>(this->one));
+}
+
+
+template<typename T>
+class StringFromUnsignedIntegerTest : public ::testing::Test
+{
+  public:
+	StringFromUnsignedIntegerTest<T>()
+	{}
+
+	T zero{0};
+	T one{1};
+};
+
+using StringFromUnsignedIntegerTestTypes = ::testing::Types<unsigned short, unsigned int, unsigned long, unsigned long long>;
+TYPED_TEST_SUITE(StringFromUnsignedIntegerTest, StringFromUnsignedIntegerTestTypes);
```

Comment:
Same as previous class.

----

Actually, if both classes are empty,  maybe there's a way to reuse the same class.

What really matters is the test function needs to be properly paired with the test type list. The test harness (class) doesn't really do anything on it's own. Though I suppose it serves as the name that ties the test function(s) to the list of test types.

Reply:
Just tried reusing an empty class. `TYPED_TEST_SUITE` requires the test suite name (the first parameter of the macro) to be unique or it throws an "redefinition; different basic types" error.